File: src/Counsel.FantasyClient/Models/AdvancedPlayerStatistics.cs
using Counsel.FantasyClient.Converters;
using Newtonsoft.Json;

namespace Counsel.FantasyClient.Models
{
    public class AdvancedPlayerStatistics
    {
        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? Carries { get; set; }

        [JsonProperty("fanPtsAgainstOpponentPts")]
        [JsonConverter(typeof(StringToNullableConverter<double>))]
        public double? FanPointsAgainstOpponentPoints { get; set; }

        [JsonProperty("fanPtsAgainstOpponentRank")]
        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? FanPointsAgainstOpponentRank { get; set; }

        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? ReceptionPercentage { get; set; }

        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? Receptions { get; set; }

        /// <summary>
        /// Number of red zone targets and touches when down is goal-to-go
        /// </summary>
        /// <value>The redzone goal to go.</value>
        [JsonProperty("redzoneG2g")]
        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? RedzoneGoalToGo { get; set; }

        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? RedzoneTargets { get; set; }

        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? RedzoneTouches { get; set; }

        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? Targets { get; set; }

        [JsonConverter(typeof(StringToNullableConverter<int>))]
        public int? Touches { get; set; }
    }
}

File: src/Counsel/FantasyClient/AdvancedPlayerStatistics.swift
//
//  AdvancedPlayerStatistics.swift
//  Counsel
//
//  Created by Han Hossain on 6/9/19.
//  Copyright © 2019 Han Hossain. All rights reserved.
//

import Foundation

struct AdvancedPlayerStatistics {
	let carries: Int?
	let fanPointsAgainstOpponentPoints: Double?
	let fanPointsAgainstOpponentRank: Double?
	let receptionPercentage: Int?
	let receptions: Int?
	let redzoneGoalToGo: Int?
	let redzoneTargets: Int?
	let redzoneTouches: Int?
	let targets: Int?
	let touches: Int?
}

extension AdvancedPlayerStatistics {
	init(json: [String: Any]) throws {
		if let carries = json["Carries"] as? String {
			self.carries = Int(carries)
		} else {
			carries = nil
		}
		
		if let fanPointsAgainstOpponentPoints = json["FanPtsAgainstOpponentPts"] as? String {
			self.fanPointsAgainstOpponentPoints = Double(fanPointsAgainstOpponentPoints)
		} else {
			fanPointsAgainstOpponentPoints = nil
		}
		
		if let fanPointsAgainstOpponentRank = json["FanPtsAgainstOpponentRank"] as? String {
			self.fanPointsAgainstOpponentRank = Double(fanPointsAgainstOpponentRank)
		} else {
			fanPointsAgainstOpponentRank = nil
		}
		
		if let receptionPercentage = json["ReceptionPercentage"] as? String {
			self.receptionPercentage = Int(receptionPercentage)
		} else {
			receptionPercentage = nil
		}
		
		if let receptions = json["Receptions"] as? String {
			self.receptions = Int(receptions)
		} else {
			receptions = nil
		}
		
		if let redzoneGoalToGo = json["RedzoneG2g"] as? String {
			self.redzoneGoalToGo = Int(redzoneGoalToGo)
		} else {
			redzoneGoalToGo = nil
		}
		
		if let redzoneTargets = json["RedzoneTargets"] as? String {
			self.redzoneTargets = Int(redzoneTargets)
		} else {
			redzoneTargets = nil
		}
		
		if let redzoneTouches = json["RedzoneTouches"] as? String {
			self.redzoneTouches = Int(redzoneTouches)
		} else {
			redzoneTouches = nil
		}
		
		if let targets = json["Targets"] as? String {
			self.targets = Int(targets)
		} else {
			targets = nil
		}
		
		if let touches = json["Touches"] as? String {
			self.touches = Int(touches)
		} else {
			touches = nil
		}
	}
}
